app/models/mixins/purging_mixin.rb

```diff
@@ -46,6 +60,57 @@ def purge_by_date(older_than, window = nil, &block)
       total
     end
 
+    def purge_count_by_remaining(remaining)
+      purge_ids_for_remaining(remaining).size
+    end
+
+    # @param [Integer] remaining number of records per resource to keep (remains in table)
+    # @param [Integer] window number of records to delete in a batch
+    # @return [Integer] number of records deleted
+    def purge_by_remaining(remaining, window = nil, &block)
+      _log.info("Purging #{table_name.humanize} older than last #{remaining} results...")
+      total = purge_in_batches(purge_ids_for_remaining(remaining), window || purge_window_size, &block)
+      _log.info("Purging #{table_name.humanize} older than last #{remaining} results...Complete - Deleted #{total} records")
+      total
+    end
+
+    # Private:  The ids to purge if we want to keep a fixed number of records
+    # for each resource. Newer records (with a higher id/ lower rank) are kept.
+    #
+    # example:
+    #
+    # So given the following DriftStates:
+    #
+    # {id: 1, resource_type: 'foo', resource_id: 1}  # Rank = 2
+    # {id: 2, resource_type: 'foo', resource_id: 1}  # Rank = 1
+    # {id: 3, resource_type: 'foo', resource_id: 2}  # Rank = 2
+    # {id: 4, resource_type: 'foo', resource_id: 2}  # Rank = 1
+    # {id: 5, resource_type: 'bar', resource_id: 6}  # Rank = 1
+    # {id: 6, resource_type: 'baz', resource_id: 4}  # Rank = 1
+    #
+    # For a "remaining" value of `1`, id 1 and 3 would then be up for deletion.
+    #
+    # @param [Integer] remaining number of records per resource to keep (remains in table)
+    # @return [Scope] Records to be deleted
+    def purge_ids_for_remaining(remaining)
+      # NOTE: `as(table_name)` fixes a bug with pluck(:id)
```

Comment:
Is there an open issue for this bug?  If so, can you link to it here so we can know if it's fixed in the future and know to undo the hack?  If it's a hack can you also use the `# HACK: ` prefix instead of NOTE?